### PR TITLE
Improves json validation readme

### PR DESCRIPTION
**Pull request by jcchavezs**

### CODE REVIEW

File: examples/json_validation/README.md [hunk 1]
```diff
@@ -54,6 +57,7 @@ This example details deploying to a kind cluster running the Istio httpbin sampl
 kind create cluster
 
 # Install Istio and the httpbin sample app
+
 istioctl install --set profile=demo -y
```
Comment (jcchavezs) [inline examples/json_validation/README.md:61]: @liamawhite could you please test from here on? I am in trouble to run this in my M1 mac now.

**Review by mathetake — COMMENTED** [review]
you want to add an e2e test in https://github.com/tetratelabs/proxy-wasm-go-sdk/blob/main/e2e/e2e_test.go ?

File: examples/json_validation/README.md [hunk 2]
```diff
@@ -54,10 +57,11 @@ This example details deploying to a kind cluster running the Istio httpbin sampl
...
 kubectl label namespace default istio-injection=enabled
-kubectl apply -f samples/httpbin/httpbin.yaml
-kubectl apply -f samples/httpbin/httpbin-gateway.yaml
+kubectl apply -f https://github.com/istio/istio/blob/48cdd8/samples/httpbin/httpbin.yaml
```
Comment (jcchavezs) [inline examples/json_validation/README.md:63]: Files come from https://github.com/istio/istio/pull/38114

**Comment by jcchavezs** [comment]
> you want to add an e2e test in https://github.com/tetratelabs/proxy-wasm-go-sdk/blob/main/e2e/e2e_test.go ?

Sounds good to me.

**Comment by jcchavezs** [comment]
Done @mathetake

File: examples/json_validation/README.md [hunk 3]
```diff
@@ -7,20 +7,24 @@ If not, the wasm plugin ceases the further process of the request and returns 40
 
 `envoy.yaml` is the example envoy config file that you can use for running the wasm plugin
 with standalone Envoy.
-The following make rule will run the wasm plugin in the version of Envoy used in the Istio 1.9 sidecar.
 
 Envoy listens on `localhost:18000`, responding to any requests with static content "hello from server".
 However, the wasm plugin also runs to validate the requests' payload.
+
+```bash
+make run name=json_validation
```
Comment (jcchavezs) [inline examples/json_validation/README.md:15]:
```suggestion
make -C ../.. run name=json_validation
```

File: e2e/e2e_test.go [hunk 4]
```diff
@@ -309,3 +310,33 @@ func Test_vm_plugin_configuration(t *testing.T) {
...
+	stdErr, kill := startEnvoy(t, 8001)
+	defer kill()
+
+	require.Eventually(t, func() bool {
+		req, _ := http.NewRequest("GET", "http://localhost:18000", nil)
+		res, err := http.DefaultClient.Do(req)
+		if err != nil {
+			return false
+		}
+		_, err = io.Copy(ioutil.Discard, res.Body)
```
Comment (mathetake) [inline e2e/e2e_test.go:324]:
```suggestion
		defer res.Body.Close()
```

File: e2e/e2e_test.go [hunk 5]
```diff
@@ -309,3 +310,33 @@ func Test_vm_plugin_configuration(t *testing.T) {
...
+
+		jsonBody := `{"id": "abc123", "token": "xyz456"}`
+
+		req, _ = http.NewRequest("POST", "http://localhost:18000", strings.NewReader(jsonBody))
+		req.Header.Add("Content-Type", "application/json")
+		res, err = http.DefaultClient.Do(req)
+		if err != nil {
+			return false
+		}
+		_, err = io.Copy(ioutil.Discard, res.Body)
```
Comment (mathetake) [inline e2e/e2e_test.go:336]:
```suggestion
		defer res.Body.Close()
```

**Review by mathetake — APPROVED** [review]

**Pull request merged**